### PR TITLE
Implement timers called out in #432

### PR DESCRIPTION
**Pull request by 0awful**
in #432 a proposed solution was offered. I've implemented it on my machine and it works. Here's that change.

`_show_start_button` was used to make cargo warnings go away.

### CODE REVIEW

File: examples/dodge-the-creeps/rust/src/hud.rs [hunk 1]
```diff
@@ -26,12 +26,17 @@ impl Hud {
...
 
+        let mut timer = self.base.get_tree().unwrap().create_timer(2.0).unwrap();
+        timer.connect("timeout".into(), self.base.callable("_show_start_button"));
+    }
+
+    #[func]
+    fn _show_start_button(&mut self) {
         let mut message_label = self.base.get_node_as::<Label>("MessageLabel");
-        message_label.set_text("Dodge the\nCreeps!".into());
+        message_label.set_text("Dodge The Creeps".into());
```
Comment (Bromeon) [inline examples/dodge-the-creeps/rust/src/hud.rs:36]: Why this change? The line break is for stylistic reasons and exists [in the original](https://github.com/godotengine/godot-demo-projects/blob/master/2d/dodge_the_creeps/HUD.gd#L14), as well.

File: examples/dodge-the-creeps/rust/src/hud.rs [hunk 2]
```diff
@@ -26,12 +26,17 @@ impl Hud {
     pub fn show_game_over(&self) {
         self.show_message("Game Over".into());
 
+        let mut timer = self.base.get_tree().unwrap().create_timer(2.0).unwrap();
+        timer.connect("timeout".into(), self.base.callable("_show_start_button"));
+    }
+
+    #[func]
+    fn _show_start_button(&mut self) {
```
Comment (Bromeon) [inline examples/dodge-the-creeps/rust/src/hud.rs:34]: In Rust, underscores at the beginning of symbols means "unused", while in GDScript it means "private" (by convention). Since we have better means to convey visibility, we don't generally use this convention.

File: examples/dodge-the-creeps/rust/src/hud.rs [hunk 3]
```diff
@@ -26,12 +26,17 @@ impl Hud {
...
         let mut message_label = self.base.get_node_as::<Label>("MessageLabel");
-        message_label.set_text("Dodge the\nCreeps!".into());
+        message_label.set_text("Dodge The Creeps".into());
         message_label.show();
-
         let mut button = self.base.get_node_as::<Button>("StartButton");
-        button.show();
+        button.show()
```
Comment (Bromeon) [inline examples/dodge-the-creeps/rust/src/hud.rs:39]: Why remove the semicolon?

**Review by Bromeon — COMMENTED** [review]
Thanks! Added few comments.

There's some stuff in CI I need to fix, will run once re-enabled.

File: examples/dodge-the-creeps/rust/src/hud.rs (hunk 1 above)
Comment (0awful) [reply examples/dodge-the-creeps/rust/src/hud.rs:36]: Carryover from my implementation. I set a different screen size and didn't catch a difference. Swapping back and then squashing into a single commit.

File: examples/dodge-the-creeps/rust/src/hud.rs (hunk 3 above)
Comment (0awful) [reply examples/dodge-the-creeps/rust/src/hud.rs:39]: It doesn't need to be removed. I'll put it back and squash into single commit.

File: examples/dodge-the-creeps/rust/src/hud.rs (hunk 2 above)
Comment (Bromeon) [reply examples/dodge-the-creeps/rust/src/hud.rs:34]: That's strange, because the `#[func]` macro should expsnd to code that _does_ use the function. We have dozens of tests doing this, and even clippy is fine with it.

What warnings do you get?

Comment (0awful) [reply examples/dodge-the-creeps/rust/src/hud.rs:34]: Looks like my editor was drunk. I'll go with non `_` naming. I'll clean this up per contrib guidelines and reopen.

**Comment by 0awful** [comment]
Yay! Its open.

Now to the weirdness. Per #521. Cargo gives this build error on this code when attempting to compile manually.
```
error[E0599]: no method named `create_timer` found for struct `godot::prelude::Gd<SceneTree>` in the current scope
  --> examples/dodge-the-creeps/rust/src/hud.rs:29:55
   |
29 | ....unwrap().create_timer(2.0).unwrap();
   |              ^^^^^^^^^^^^ method not found in `Gd<SceneTree>`

```
This same code works in another project, but not for this example.

**Comment by GodotRust** [comment]
API docs are being generated and will be shortly available at: https://godot-rust.github.io/docs/gdext/pr-514

**Comment by GodotRust** [comment]
API docs are being generated and will be shortly available at: https://godot-rust.github.io/docs/gdext/pr-514

**Comment by 0awful** [comment]
`cargo clean` got it. Now it works.

**Comment by Bromeon** [comment]
> Cargo gives this build error on this code when attempting to compile manually.

I got that error too when testing. Was first suspecting some classes to be missing in `SELECTED_CLASSES`, but it looks like both `SceneTree` and its base `MainLoop` are already in. I added specific CI for examples, will rebase this and see what it says...

**Comment by Bromeon** [comment]
Problem was that `SceneTreeTimer` was not in `SELECTED_CLASSES`, and so any methods having this type in the signature are not generated. Fixed it on my end.
